src/allocation_engine/interval_tree.rs

```diff
@@ -598,7 +699,7 @@ mod tests {
             .unwrap();
 
         assert_eq!(
-            tree.search_superset(&Range::new(0x100, 0x100).unwrap()),
+            tree.search_superset(&Range::new(0x100, 0x101).unwrap()),
```

Comment:
Can you add some more documentation for these tests? I don't quite understand why you changed the values :)

Reply:
Because Range does not support having the start equal with the end anymore.